src/eth_client.rs

```diff
@@ -0,0 +1,146 @@
+use bytes::Bytes;
+use ethereum_types::{Address, U256};
+/// This client is inspired in ethrex EthClient https://github.com/lambdaclass/ethrex/blob/main/crates/l2/utils/eth_client/mod.rs
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+
+use crate::errors::EthClientError;
+
+/// Request ID for the RPC
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum RpcRequestId {
+    Number(u64),
+    String(String),
+}
+
+/// Response for a successful RPC request
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct RpcSuccessResponse {
+    pub(crate) id: RpcRequestId,
+    pub(crate) jsonrpc: String,
+    pub(crate) result: Value,
+}
+
+/// Metadata for an RPC error
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct RpcErrorMetadata {
+    pub(crate) code: i32,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) data: Option<String>,
+    pub(crate) message: String,
+}
+
+/// Response for an RPC error
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcErrorResponse {
+    pub(crate) id: RpcRequestId,
+    pub(crate) jsonrpc: String,
+    pub(crate) error: RpcErrorMetadata,
+}
+
+impl std::fmt::Display for RpcErrorResponse {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "RpcErrorResponse: {:?}", self)
+    }
+}
+
+/// Response for an RPC request
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+pub(crate) enum RpcResponse {
+    Success(RpcSuccessResponse),
+    Error(RpcErrorResponse),
+}
+
+/// Request for the RPC
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct RpcRequest {
+    pub(crate) id: RpcRequestId,
+    pub(crate) jsonrpc: String,
+    pub(crate) method: String,
+    pub(crate) params: Option<Vec<Value>>,
+}
+
+/// Client for interacting with an Ethereum node
+#[derive(Debug, Clone)]
+pub(crate) struct EthClient {
+    client: Client,
```

Comment:
```suggestion
    client: reqwest::Client,
```

Reply:
I prefer to have the module name there when it's short, otherwise I need to go to the import section at the top to figure out what this Client is.

Reply:
addressed here https://github.com/lambdaclass/eigenda-client-rs/pull/4